python/tk_multi_loader/dialog.py

```diff
@@ -1265,6 +1269,110 @@ def _load_entity_presets(self):
         # data has properly arrived in the model.
         self._on_home_clicked()
 
+    def _setup_hierarchy_model(self, app, root):
+        """
+        Create the model and proxy model required by a hierarchy type configuration setting.
+
+        :param app: :class:`Application`, :class:`Engine` or :class:`Framework` bundle instance
+                    associated with the loader.
+        :param root: The path to the root of the Shotgun hierarchy to display.
+        :return: Created `(model, proxy model)`.
+        """
+
+        # Resolve any magic tokens in the root.
+        if app.context.project:
```

Comment:
These are two very different configuration settings:
- The "root" one is simple and involves replacing the "magic token" in the root path.
- The "filters" one is complex and involves replacing the "magic tokens" in a list of of lists.
Centralizing into a single method would not bring any benefit and would make the code harder to understand and maintain.